File: features/offices/src/main/java/com/nivasafinance/features/offices/entity/Office.java
package com.nivasafinance.features.offices.entity;

import com.nivasafinance.common.audit.AuditableEntity;
import com.nivasafinance.common.dto.AddressData;
import io.hypersistence.utils.hibernate.type.json.JsonType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.annotations.Type;
import org.hibernate.type.SqlTypes;

@Entity
@Table(name = "n_office")
@NoArgsConstructor
@AllArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
public class Office extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "key", nullable = false, unique = true)
    private String key;

    @Column(name = "code", nullable = false, unique = true)
    private String code;

    @Type(JsonType.class)
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "address_data", columnDefinition = "jsonb")
    private AddressData addressData;

    @Column(name = "parent_id")
    private Long parentId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id", insertable = false, updatable = false)
    private Office parent;

    @Column(name = "is_active")
    private Boolean isActive;
}

File: features/offices/src/main/java/com/nivasafinance/features/offices/entity/Office.kt
package com.nivasafinance.features.offices.entity

import audit.AuditableEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.util.UUID

@Entity
@Table(name = "offices")
class Office(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "UUID")
    val id: UUID? = null,

    @Column(name = "name", nullable = false)
    var name: String = "",

    @Column(name = "key", nullable = false)
    var key: String = "",

    @Column(name = "code", nullable = false)
    var code: String = "",

    @Column(name = "address_id")
    var addressId: UUID? = null,

    @Column(name = "parent_id")
    var parentId: UUID? = null,

    @ManyToOne
    @JoinColumn(name = "parent_id", insertable = false, updatable = false)
    var parent: Office? = null,
) : AuditableEntity()
